clojure/src/info/lotharschulz/armstrong_numbers/armstrong_numbers.clj

```diff
@@ -0,0 +1,38 @@
+(ns info.lotharschulz.armstrong_numbers.armstrong_numbers
+  (:import [java.Math]))
+
+(defn get-number-of-digits [^Integer n]
+  {:pre [(not (nil? n)) (integer? n)]}
+  "Returns the :getNumberOfDigits of n if n is an integer and greater than -1"
+  (if (zero? n)
+    1
+    (if (pos? n )
+      (+ (int (Math/log10 n )) 1)
+      nil)))
+
+(defn get-digits [^Integer n]
+  {:pre [(not (nil? n)) (integer? n)]}
+  "return number of digits of given int n"
+    (if (zero? n)
+      [0]
+      (if (pos? n)
+        (into [] (map #(Character/getNumericValue %) (str n)))
+        [])))
+
+(defn is-armstrong-number [^Integer n]
+  {:pre [(not (nil? n)) (integer? n)]}
+  "Returns true if the given number bigger than -1 is an armstrong number"
+  (if (zero? n)
+    false
+    (if (pos? n)
+      (= n (reduce + (map #(int (Math/pow % (get-number-of-digits n))) (get-digits n))))
+      false)))
+
+(defn generate-armstrong-numbers [^Integer from, ^Integer to]
+  {:pre [(not (nil? from)) (integer? from) (pos? from) (not (nil? to)) (integer? to) (pos? to) ]}
```

Comment:
And if you call `(is-armstrong-number "2")`, you will immediately know what's wrong anyway, there will be `java.lang.ClassCastException`.